File: CQRS-write/cqrswrite-infrastructure/src/main/java/com/example/cqrswrite/domain/feed/presentation/FeedWebAdapter.java
package com.example.cqrswrite.domain.feed.presentation;

import com.example.cqrswrite.domain.feed.dto.request.CreateFeedRequest;
import com.example.cqrswrite.domain.feed.presentation.dto.request.CreateFeedWebRequest;
import com.example.cqrswrite.domain.feed.usecase.CreateFeedUseCase;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

@RequiredArgsConstructor
@RequestMapping("/feeds")
@RestController
public class FeedWebAdapter {

    private final CreateFeedUseCase createFeedUseCase;

    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping
    public void createFeed(@RequestBody @Valid CreateFeedWebRequest request) {
        createFeedUseCase.execute(
                CreateFeedRequest.builder()
                        .title(request.getTitle())
                        .content(request.getContent())
                        .build()
        );
    }
}

File: CQRS-read/cqrsread-infrastructure/src/main/kotlin/com/example/cqrsread/domain/feed/presentation/FeedWebAdapter.kt
package com.example.cqrsread.domain.feed.presentation

import com.example.cqrsread.domain.feed.dto.response.QueryFeedsResponse
import com.example.cqrsread.domain.feed.model.Feeds
import com.example.cqrsread.domain.feed.usecase.QueryFeedsUseCase
import kotlinx.coroutines.flow.Flow
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/feeds")
@RestController
class FeedWebAdapter(
    private val queryFeedsUseCase: QueryFeedsUseCase,
) {

    @GetMapping
    suspend fun queryFeeds(): Flow<Feeds> =
        queryFeedsUseCase.execute()
}